benchmarks/sift/datagen.py

```diff
@@ -31,23 +31,40 @@ def main():
         help="Max number of records per data file",
         metavar="NUM",
     )
+    parser.add_argument(
+        "-d",
+        "--dimension",
+        type=int,
+        default=128,
+        help="Dimension of the vector",
+        metavar="NUM",
+    )
+    parser.add_argument(
+        "-n",
+        "--num-vectors",
+        type=int,
+        default=1000000,
+        help="Number of vectors to generate",
+        metavar="NUM",
+    )
     args = parser.parse_args()
 
     with open(args.data, mode="rb") as fobj:
         buf = fobj.read()
-        data = np.array(struct.unpack("<128000000f", buf[4 : 4 + 4 * 1000000 * 128]))
+        data = np.array(struct.unpack(f"<{args.dimension * args.num_vectors}f", 
+                                      buf[4 : 4 + 4 * args.num_vectors * args.dimension]))
```

Comment:
guess for the benchmark dataset size, we will not have any memoryerror concern

Reply:
no it's only 1M vectors

Reply:
Most public benchmarks are of this size and dimensionality, so it shouldn't be a concern as long as this code is used for those benchmarks.